src/smc-webapp/r_misc.cjsx

```diff
@@ -921,16 +908,21 @@ exports.HTML = rclass
                 html = require('./misc_page').sanitize_html_safe(@props.value, @props.post_hook)
             else
                 html = require('./misc_page').sanitize_html(@props.value, true, true, @props.post_hook)
+
+            if @props.auto_render_math
+                @_needs_mathjax = true
+
+                if not @props.only_mathjax
+                    {html, is_complete} = math_katex.render(html)
+                    @_needs_mathjax = not is_complete
+
             {__html: html}
         else
             {__html: ''}
 
 
     render: ->
-        # the random key is the whole span (hence the html) does get rendered whenever
```

Comment:
Why did you delete this -- I even left a comment about why it's needed... ;-)

Reply:
I somehow read the next line as "Otherwise does NOT render when the value changes" and I tested that lol